types/field_type.go

```diff
@@ -261,8 +261,8 @@ func DefaultTypeForValue(value interface{}, tp *FieldType, char string, collate
 		tp.Flag |= mysql.UnsignedFlag
 		SetBinChsClnFlag(tp)
 	case BinaryLiteral:
-		tp.Tp = mysql.TypeBit
-		tp.Flen = len(x) * 8
```

Comment:
This change is very bottomed. We'd better evaluate its effect carefully.

Reply:
According to this [PR](https://github.com/pingcap/tidb/pull/4415), BinaryLiteral should be TypeVarString. So i think this is correct. But we do need more test to verify it's really OK.